analyzers/src/SonarAnalyzer.Common/Rules/DoNotCallInsecureSecurityAlgorithmBase.cs

```diff
@@ -23,63 +23,88 @@
 using System.Collections.Immutable;
 using System.Linq;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Helpers;
 
-namespace SonarAnalyzer.Rules.CSharp
+namespace SonarAnalyzer.Rules
 {
-    public abstract class DoNotCallInsecureSecurityAlgorithm : SonarDiagnosticAnalyzer
+    public abstract class DoNotCallInsecureSecurityAlgorithmBase<TSyntaxKind, TInvocationExpressionSyntax, TObjectCreationExpressionSyntax, TArgumentListSyntax, TArgumentSyntax>
+        : SonarDiagnosticAnalyzer
+        where TSyntaxKind : struct
+        where TInvocationExpressionSyntax : SyntaxNode
+        where TObjectCreationExpressionSyntax : SyntaxNode
+        where TArgumentListSyntax : SyntaxNode
+        where TArgumentSyntax : SyntaxNode
     {
-        internal abstract ImmutableArray<KnownType> AlgorithmTypes { get; }
         protected abstract ISet<string> AlgorithmParameterlessFactoryMethods { get; }
         protected abstract ISet<string> AlgorithmParameterizedFactoryMethods { get; }
         protected abstract ISet<string> FactoryParameterNames { get; }
+        protected abstract TSyntaxKind ObjectCreation { get; }
+        protected abstract TSyntaxKind Invocation { get; }
+        protected abstract ILanguageFacade LanguageFacade { get; }
```

Comment:
To align with the other PRs:
```suggestion
        protected abstract ILanguageFacade Language { get; }
```